Tests/GitHubContents.tests.ps1

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+.Synopsis
+   Tests for GitHubContents.ps1 module
+#>
+
+# This is common test code setup logic for all Pester test files
+$moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
+. (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Common.ps1')
+
+try {
+    # Define Script-scoped, readonly, hidden variables.
+    @{
+        repoGuid       = [Guid]::NewGuid().Guid
+        readmeFileName = "README.md"
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    # Need this twice since we need to use the above var here
+    @{
+        htmlOutput = "<div id=`"file`" class=`"md`" data-path=`"README.md`"><article class=`"markdown-body entry-content p-5`" itemprop=`"text`"><h1><a id=`"user-content-$repoGuid`" class=`"anchor`" aria-hidden=`"true`" href=`"#$repoGuid`"><svg class=`"octicon octicon-link`" viewBox=`"0 0 16 16`" version=`"1.1`" width=`"16`" height=`"16`" aria-hidden=`"true`"><path fill-rule=`"evenodd`" d=`"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z`"></path></svg></a>$repoGuid</h1></article></div>"
```

Comment:
I feel like this test is likely a little delicate because you're dependent on both the content of the default README.md that GitHub autogenerates to never change, as well as the way that it generates HTML from the MD file.  Ideally, we would make this test more robust by uploading our own file, and then downloading it and verifying that it round-tripped correctly.  Unfortunately, to do that would require the implementation of the [Create or update a file](https://developer.github.com/v3/repos/contents/#create-or-update-a-file) API which is beyond what you were intending to implement.

This test is solid as-is, but will likely need to be updated once that other API gets implemented in the module.

Reply:
Yes i totally agree! My plan was on the next iteration to build the create/update file API and then update these tests accordingly but wanted to get this functionality done first and then update these tests later once that functionality was implemented.